Respin draft around integration properties (#30)
tls-attestation/use-cases-and-properties: draft-mihalcea-seat-use-cases.md

Reviewer: I propose a different title: Attested Secure Channel Establishment: Use Cases and Requirements

Author: Done

Reviewer 2: ```suggestion
  ID-Crisis:
    title: "Identity Crisis in Confidential Computing: Formal Analysis of Attested TLS"
    date: November 2025,
    target: https://www.researchgate.net/publication/398839141_Identity_Crisis_in_Confidential_Computing_Formal_Analysis_of_Attested_TLS
    author:
      - ins: M. U. Sardar
      - ins: M. Moustafa
      - ins: T. Aura
```

Reviewer: I suggest adding a new section: 

Identity Binding Requirement

The authenticated identity of a peer (e.g., as established via PKI-based authentication in TLS) has to be cryptographically bound to the attested TEE. As discussed in {{ID-Crisis}}, failure to establish such binding can lead to diversion or identity misbinding attacks, where the authenticated identity does not correspond to the peer that produced the attestation evidence. The attestation evidence has to be cryptographically bound to the keying material used by the peer in the secure channel establishment.

Author: Wait, is this a replacement for either connection binding, or binding to machine identifier? I'm confused because to me this seems like an alternative to connection binding:

* Evidence / AR is cryptographically bound to specific TLS connection
* Authenticated identity is cryptographically bound to specific TLS connection (via CertificateVerify)
* Evidence / AR is cryptographically bound to Authenticated identity

The 3rd step seems redundant given the first two. Or, the first step seems redundant given the 2nd and 3rd.

Reviewer: It is different from https://github.com/tls-attestation/use-cases-and-properties/blob/properties-focus/draft-mihalcea-seat-use-cases.md#attestation-credential-binding-to-the-secure-channel. 

The requirement will be to address the problem discussed in {{ID-Crisis}} and https://datatracker.ietf.org/doc/draft-reddy-rats-key-binding/

Reviewer 2: Discussed in meeting:
1. Private key is stored in TEE and not in rich OS. 
2. Private key is non-exportable.

---

> title: "Attested Secure Channel Establishment: Use Cases and Requirements"

Reviewer 2: I think the previous title was better.

--- abstract

This document outlines desirable properties for integrating remote attestation
(RA) capabilities with secure channel establishment protocols, along with the
ecosystem use cases from which they were derived. The initial focus is on the

> Transport Layer Security (TLS) v1.3 Handshake protocol. Traditional peer

Reviewer: I would suggest using (D)TLS 1.3 and adding a note in the terminology section that “TLS” refers to both DTLS and TLS unless explicitly specified.

Author: Done

Party.

### Integration with Authentication

The solution should support using RA in conjunction with traditional PKI-based
authentication (e.g., X.509 certificates). This provides two independent pillars
of trust: trustworthiness (from RA) and network identity (from PKI).

### Privacy Preservation

The solution must limit its effect on the privacy properties of the secure
channel establishment protocol. Evidence can contain highly specific, unique
information about a device's hardware and software, which could be used as an
advanced tracking mechanism. The design must consider how to minimize this
leakage.

### Performance and Efficiency

The introduction of attestation should not add prohibitive latency or overhead
to the connection establishment process. Multiple additional round-trips or very
large payloads in the initial handshake should be minimized so that the
solution remains practical.

## Use-Case-Specific Properties

These properties are only required for specific use-cases.

### Cryptographic Binding to Machine Identifier

Evidence and attestation results can be securely linked to location information
about the Attester who produced it to prevent diversion attacks

> {{Meeting-122-TLS-Slides}}. This property is especially important in

Reviewer 2: ```suggestion
{{ID-Crisis}}. This property is especially important in
```

prompt template) can change independently of the binary/image and usually
more frequently than typical platform TCB updates {{AI-agents}}.

# Integration Properties

Following the model used in {{RFC7478}}, the primary output of the use cases
collected in this document is a set of protocol integration properties. These
properties guide the design of RA integration with secure channel protocols.
They are split into generic requirements that must hold for all deployments and
use-case-specific requirements that arise in particular scenarios. The use
cases in the next section explicitly reference the relevant properties.

## Generic Properties

These properties are necessary for any valid protocol design, regardless of any
use-cases they might be intended for.

> ### Cryptographic Binding to Communication Channel

Reviewer 3: We should make the subject (i.e., “Attestation Credentials”) explicit in the title.

To avoid a mouthful, perhaps: _“Attestation Credential binding to the Secure Channel”_.

Author: Done!

The attestation Evidence or Attestation Result is cryptographically bound to the
specific secure channel instance (e.g., the TLS connection). This prevents
replay and relay attacks where an attacker presents valid, but old or unrelated
Evidence from a different connection or context. This binding is paramount for

> all use cases.

Reviewer 3: Passport is the 🐘 in this subsection and the next one … It’s not clear to me how these can be made compatible with _“Attestation Model Flexibility”_.

Author: Indeed. Couple with that with a clear performance penalty for background check and I'm not sure where this whole thing leaves us.

I think it's likely we'll end up with a scheme along the lines of: "AR contains some claim about the TIK, and the TIK is the thing that binds everything together." The counter-argument would be 'what if the TIK_priv is leaked?' but that sort of attack breaks everything anyway.

Reviewer: If TIK_priv is compromised, it is identity theft.

Reviewer 2: Attestation Results should be bound to the TLS connection as per SEAT charter:

> The attested D(TLS) protocol extension will also describe a minimum
subset of properties that the attested state must convey in order
to bind the Evidence and Attestation Results to the TLS connection.

Attestation Results produced before TLS connection are not within the scope.

### Attestation Credential Freshness

The Relying Party can verify that the Evidence or Attestation Result it receives
was freshly generated by the Attester for the current connection. State is

> transient, and credentials from a previous connection may no longer be valid.

Reviewer 3: The sentence _“State is transient […]”_ sounds a bit disconnected.  Is there anything we could add to link it to the rest of the paragraph?

Author: Done. I've actually changed the freshness section quite thoroughly. The previous version was leaning way too much in the direction of equating freshness and channel binding (e.g., "freshly generated by the Attester _for the current connection_").

models they support. This enables interoperability and allows for graceful
fallback for endpoints that do not support RA.

### Attestation Model Flexibility

The solution supports both the Background Check and Passport models as defined
in the RATS architecture {{RFC9334}}. The Background Check model is essential
for deployments requiring maximum freshness, while the Passport model is better
suited for performance, scalability, and scenarios where the Verifier may be
offline or unreachable by the Relying Party.

### Integration with Authentication

The solution supports using RA in conjunction with traditional PKI-based
authentication (e.g., X.509 certificates). This provides two independent pillars
of trust: trustworthiness (from RA) and network identity (from PKI).

### Privacy Preservation

The solution must limit its effect on the privacy properties of the secure
channel establishment protocol. Evidence can contain highly specific, unique
information about a device's hardware and software, which could be used as an
advanced tracking mechanism. The design must consider how to minimize this
leakage, especially when a third-party Verifier is involved in the protocol

> exchange.

Reviewer 3: I'm wondering whether this "especially […]" is really a requirement, though.

It seems purely a requirement concerning the RA protocol, rather than the _integration_ between RA and secure channel establishment.

Author: Right, the integration protocol design should be agnostic of that, since you can't know a priori. Removed.

### Performance and Efficiency

The introduction of attestation should not add prohibitive latency or overhead
to the connection establishment process. Multiple additional round-trips or very
large payloads in the initial handshake should be minimized so that the
solution remains practical.

## Use-Case-Specific Properties

These properties are only required for specific use-cases.

### Cryptographic Binding to Machine Identifier

Evidence is cryptographically bound to an identifier provided to the machine by

> the infrastructure provider to prevent diversion attacks {{Meeting-122-TLS-Slides}}.

Reviewer 3: This should be presented as a requirement rather than a solution.

We should phrase it the other way around, for example:

_"Evidence can be securely linked to location information about the Attester who produced it to prevent diversion attacks.”_

Author: Ack, done.

prompt template) can change independently of the binary/image and usually
more frequently than typical platform TCB updates {{AI-agents}}.

# Integration Properties

Following the model used in {{RFC7478}}, the primary output of the use cases
collected in this document is a set of protocol integration properties. These

> properties guide the design of RA integration with secure channel protocols.

Reviewer: ```suggestion
properties guide the design of RA integration with TLS.
```

Author: The wording here was meant to be non-specific, as we could repurpose the contents for other protocols in the future

Reviewer: I see your point but scope of the WG is TLS; we could say something like 

These properties guide the design of RA integration with TLS and can be extended to other security protocols in the future.

Author: Thanks, done!

They are split into generic requirements that must hold for all deployments and
use-case-specific requirements that arise in particular scenarios. The use
cases in the next section explicitly reference the relevant properties.

## Generic Properties

These properties are necessary for any valid protocol design, regardless of any
use-cases they might be intended for.

### Attestation Credential binding to the Secure Channel

> Attestation Evidence or Attestation Results are cryptographically bound to the

Reviewer: It may not be possible to bind attestation results to the TLS connection.

Author: Yeah, same as with Thomas' comment, I agree. Guess the question is: where does that leave us? Do we just say "Evidence is always bound, ARs are only bound when they can be"?

Clearly we need to decide and define what is good enough as an alternative for ARs, otherwise RPs won't be able to trust that they represent the peer.

Reviewer: ARs as per RATS is passport model, TLS binding is not possible; it has an advantage of avoiding communication with verifier and the checks in the verifier to validate the evidence; if there is an deployment which wants to use it at the cost of losting the TLS binding; it is delibrate choice with weak security properties.

Author: > TLS binding is not possible

I don't think that's technically impossible, just very bad efficiency-wise. The attester could get a new AR per connection, from a Verifier, whenever that AR is needed. It's no different from just getting evidence, just that you have to also pass that evidence to the Verifier (who has to also reflect the binder in the AR).

But I agree that it's unlikely for this approach to actually be used (unless the Verifier is on the same machine and the latency is basically 0).

Reviewer 3: > But I agree that it's unlikely for this approach to actually be used (unless the Verifier is on the same machine and the latency is basically 0).

I agree it’s unlikely.  But, as a data point, this is what draft-ietf-acme-rats currently does.  (That said, it may change in the future.)

Reviewer: I suggest keeping it simple and aligned with RATS. The attestation results follow a passport model: they are obtained once and can be used across multiple TLS connections.

Reviewer 2: Discussion in meeting:

Attestation Results should be bound to the TLS connection as per SEAT charter:

> The attested D(TLS) protocol extension will also describe a minimum
subset of properties that the attested state must convey in order
to bind the Evidence and Attestation Results to the TLS connection.

Attestation Results produced before TLS connection are not within the scope.

specific secure channel instance (e.g., the TLS connection). This prevents
replay and relay attacks where an attacker presents valid but old or unrelated
credentials from a different connection or context. This binding is paramount
for all use cases.

### Attestation Credential Freshness

The Relying Party can verify that the Evidence or Attestation Result it receives
was freshly generated by the Attester. Since the attester's security state can

> change at any moment, stale credentials might no longer present relevant

Reviewer: ```suggestion
change at any moment, stale evidence or attestation results might no longer present relevant
```

Author: Done!

insights. See {{Section 10 of -rats-arch}} for more details about freshness in
the context of RA.

### Negotiation and Capability Discovery

Peers have a secure mechanism to discover each other's support for RA, the
specific attestation formats they can produce or consume, and the attestation
models they support. This enables interoperability and allows for graceful
fallback for endpoints that do not support RA.

### Attestation Model Flexibility

The solution supports both the Background Check and Passport models as defined
in the RATS architecture {{RFC9334}}. The Background Check model is essential
for deployments requiring maximum freshness, while the Passport model is better
suited for performance, scalability, and scenarios where the Verifier may be

> offline or unreachable by the Relying Party.

Reviewer: ```suggestion
offline or unreachable or introduce additional latency by the Relying Party.
```

Author: Done

### Integration with Authentication

> The solution supports using RA in conjunction with traditional PKI-based

Reviewer: ```suggestion
The solution should support using RA in conjunction with traditional PKI-based
```

Author: Done

authentication (e.g., X.509 certificates). This provides two independent pillars
of trust: trustworthiness (from RA) and network identity (from PKI).

### Privacy Preservation

The solution must limit its effect on the privacy properties of the secure
channel establishment protocol. Evidence can contain highly specific, unique
information about a device's hardware and software, which could be used as an
advanced tracking mechanism. The design must consider how to minimize this
leakage.

### Performance and Efficiency

The introduction of attestation should not add prohibitive latency or overhead
to the connection establishment process. Multiple additional round-trips or very
large payloads in the initial handshake should be minimized so that the
solution remains practical.

## Use-Case-Specific Properties

These properties are only required for specific use-cases.

### Cryptographic Binding to Machine Identifier

> Evidence can be securely linked to location information about the Attester who

Reviewer: It is not clear to me how location information is sufficient to detect diversion attack !

Reviewer 3: You are right.  It’s probably “ownership” that matters most.

Author: I think it's both, but just a note: location here does not necessarily mean GPS coordinates or whatever, but can be something along the lines of "AWS eu-west-1 datacenter, room 342, rack 13".

> You are right. It’s probably “ownership” that matters most.

Even if it did matter most, you still have jurisdiction issues regarding data geofencing.

That being said, I think this is something that the whole RA process should solve, and it seems irrelevant to the SCxRA integration. I'll try to figure out if there's anything about that "location" information which goes into the SC, and if not we can scrap this property.

Reviewer: The ownership or location claims in the evidence have to be defined in RATS; it is out of scope for SEAT. I suggest we remove it for now; till we have clarity on it. 

The scope of SEAT is cryptographic binding of the Evidence to the TLS connection.

Author: Agree with Tiru here.

The one argument I knew of regarding the physical machine identity is related to how that identity is conveyed, i.e., whether we need to carry it as a separate authenticated identity as part of whatever design we choose. For example, do we need to carry (and prove possession of) a new PKI-based identity, just for the machine? Or must that be folded into the attestation credential we pass around?

Reviewer: Let's delete this section.

system/network state. Its configuration (e.g., model choice, tool enablement,
prompt template) can change independently of the binary/image and usually
more frequently than typical platform TCB updates {{AI-agents}}.
* TLS: The (Datagram) Transport Layer Security protocol. Throughout this
  document, unless specified otherwise, it refers to both TLS and DTLS.

# Integration Properties

Following the model used in {{RFC7478}}, the primary output of the use cases
collected in this document is a set of protocol integration properties. These
properties guide the design of RA integration with TLS and can be extended to
other security protocols in the future. They are split into generic requirements
that must hold for all deployments and use-case-specific requirements that arise
in particular scenarios. The use cases in the next section explicitly reference
the relevant properties.

> ## Generic Properties

Reviewer 2: It would help to distinguish the verifiable properties from the other ones.

These properties are necessary for any valid protocol design, regardless of any
use-cases they might be intended for.

> ### Attestation Credential binding to the Secure Channel

Reviewer 2: ```suggestion
### Attestation Credential binding to the Connection
```

as discussed at chartering time

Reviewer 2: Discussed in meeting: 

s/Connection/TLS connection

Attestation Evidence or Attestation Results are cryptographically bound to the
specific secure channel instance (e.g., the TLS connection). This prevents
replay and relay attacks where an attacker presents valid but old or unrelated
attestation credentials from a different connection or context. This binding is
paramount for all use cases.

### Attestation Credential Freshness

The Relying Party can verify that the Evidence or Attestation Result it receives

> was freshly generated by the Attester. Since the attester's security state can

Reviewer 2: Attestation Result is not generated by the Attester.

suited for performance, scalability, and scenarios where the Verifier may
introduce additional latency, or may be offline or unreachable by the Relying
Party.

### Integration with Authentication

The solution should support using RA in conjunction with traditional PKI-based
authentication (e.g., X.509 certificates). This provides two independent pillars
of trust: trustworthiness (from RA) and network identity (from PKI).

### Privacy Preservation

The solution must limit its effect on the privacy properties of the secure
channel establishment protocol. Evidence can contain highly specific, unique
information about a device's hardware and software, which could be used as an
advanced tracking mechanism. The design must consider how to minimize this
leakage.

### Performance and Efficiency

The introduction of attestation should not add prohibitive latency or overhead
to the connection establishment process. Multiple additional round-trips or very
large payloads in the initial handshake should be minimized so that the
solution remains practical.

## Use-Case-Specific Properties

These properties are only required for specific use-cases.

> ### Cryptographic Binding to Machine Identifier

Reviewer 2: While the research {{ID-Crisis}} focused on confidential computing, I don't see why it is not applicable to other use cases.

Evidence and attestation results can be securely linked to location information
about the Attester who produced it to prevent diversion attacks
{{ID-Crisis}}. This property is especially important in
confidential computing scenarios where policy is tied to a specific platform
instance.

> ### Runtime Attestation

Reviewer 2: I think it is a fundamental property and not use-case dependent. You could be attacked just after the establishment of the connection, even if the connection is not long-lived.

These properties are necessary for any valid protocol design, regardless of any
use-cases they might be intended for.

### Attestation Credential binding to the Secure Channel

Attestation Evidence or Attestation Results are cryptographically bound to the
specific secure channel instance (e.g., the TLS connection). This prevents
replay and relay attacks where an attacker presents valid but old or unrelated

> credentials from a different connection or context. This binding is paramount

Reviewer 3: Call me an extremist :-) but to remove any possible doubt, I’d add the qualifier “attestation” to “credentials”.

Author: That's the kind of extremism I approve of. Done.

> ### Cryptographic Binding to Machine Identifier

Reviewer 3: Should the title be “Prevention of Diversion Attacks”?

Author: I partially agree, but this would make it inconsistent with, e.g., binding to secure channel. Why not name that one "Prevention of relay attacks", and freshness as "Prevention of replay attacks".

Reviewer 2: Discussed in meeting: 

Replaced this text with the older text in the published I-D.
